bindings/python/CMakeLists.txt

```diff
@@ -1,39 +1,42 @@
 set(CMAKE_SWIG_FLAGS "-Wextra;-module;iDynTree;-threads")
 set(IDYNTREE_USES_PYTHON_VERSION "" CACHE STRING "Specify python version to use" )
 
-find_package(PythonInterp ${IDYNTREE_USES_PYTHON_VERSION} REQUIRED)
-set(IDYNTREE_USES_PYTHON_VERSION_ARGS)
-  if(NOT IDYNTREE_USES_PYTHON_VERSION)
-     set (IDYNTREE_USES_PYTHON_VERSION ${PYTHON_VERSION_STRING})
-  endif()
+find_package(Python3 ${IDYNTREE_USES_PYTHON_VERSION} COMPONENTS Interpreter Development REQUIRED)
+message(STATUS "Using Python: ${Python3_EXECUTABLE}")
 
-find_package(PythonLibs ${IDYNTREE_USES_PYTHON_VERSION} EXACT)
-include_directories(${PYTHON_INCLUDE_PATH})
-link_libraries(${PYTHON_LIBRARY})
-
-## in Windows it seems necessary to declare explictly the link_directory
-if(WIN32)
-  get_filename_component(PYTHON_DIR ${PYTHON_LIBRARY} PATH)
-  link_directories(${PYTHON_DIR})
+if(NOT IDYNTREE_USES_PYTHON_VERSION)
+    set (IDYNTREE_USES_PYTHON_VERSION ${Python3_VERSION})
```

Comment:
As `IDYNTREE_USES_PYTHON_VERSION` is a cache variable, this is either not doing anything, or in any case something of really confusing.

Reply:
Right, in light of also https://github.com/robotology/idyntree/pull/726#discussion_r479650740 I removed it.